misc-utils/uuidd.c

```diff
@@ -532,22 +542,10 @@ static void __attribute__ ((__noreturn__)) unexpected_size(int size)
 	errx(EXIT_FAILURE, _("Unexpected reply length from server %d"), size);
 }
 
-int main(int argc, char **argv)
+static void parse_options(int argc, char **argv, struct uuidd_cxt_t *uuidd_cxt,
+			  struct uuidd_options_t *uuidd_opts)
 {
-	const char	*socket_path = UUIDD_SOCKET_PATH;
-	const char	*pidfile_path = NULL;
-	const char	*err_context = NULL;
-	char		buf[1024], *cp;
-	char		str[UUID_STR_LEN];
-	uuid_t		uu;
-	int		i, c, ret;
-	int		do_type = 0, do_kill = 0, num = 0;
-	int		no_pid = 0;
-	int		s_flag = 0;
-
-	struct uuidd_cxt_t uuidd_cxt = { .timeout = 0 };
-
-	static const struct option longopts[] = {
+	const struct option longopts[] = {
```

Comment:
The "static const" means that the struct will be in read-only memory (.rodata) and not in a stack, etc.

Reply:
Note that I have doubts the compiler will keep it all in stack as there are already many const strings which probably go to read-only memory, but not sure ...